src/include/binder/binder_sherpa.h

```diff
@@ -113,11 +118,25 @@ class BinderSherpa {
    */
   bool HasCTETableName(const std::string &cte_table_name) const { return cte_table_names_.count(cte_table_name) > 0; }
 
+  /**
+   * Get a unique table serial number for this query. This will help ensure that all table aliases are unique.
+   * @return unique serial number
+   */
+  parser::alias_oid_t GetUniqueTableAliasSerialNumber() {
+    auto unique_serial_number = unique_table_alias_serial_num_++;
+    if (unique_serial_number.UnderlyingValue() == 0) {
```

Comment:
Do you think I should remove this then? I was actually a little conflicted here because I'll be very slightly slowing down every single query to check for an error condition that will likely never happen. I was just hoping branch prediction would make this a no-op.